File: JudgeGreg/Models/ChatMessage.swift
import Foundation

struct ChatMessage: Identifiable, Equatable {
    let id: UUID
    let role: Role
    let content: String
    let timestamp: Date

    enum Role {
        case user
        case judge
    }

    init(role: Role, content: String) {
        self.id        = UUID()
        self.role      = role
        self.content   = content
        self.timestamp = Date()
    }
}

File: app/src/types/ChatMessage.ts
export type MessageRole = 'user' | 'judge';

export interface ChatMessage {
  id: string;
  role: MessageRole;
  content: string;
  timestamp: Date;
  isError?: boolean;
}

// Internal API-only message types (never rendered directly in the UI)
export interface ToolUseMessage {
  id: string;
  role: 'tool_use';
  toolUseId: string;
  toolName: string;
  toolInput: Record<string, unknown>;
}

export interface ToolResultMessage {
  id: string;
  role: 'tool_result';
  toolUseId: string;
  content: string;
}

export type SessionMessage = ChatMessage | ToolUseMessage | ToolResultMessage;
